### PR TITLE
Creating an upgrade process

### PR DESCRIPTION
**Pull request by jhinkey**
Hey Cristina!

As SME, please review the tutorial on creating an upgrade process for your app. In the intro, I included a draft diagram to conceptualize the registrator and upgrade steps. After any changes you suggest for it, I'll send it to Juan Hidalgo for the design team to refine.

Also, in comments near the end of the tutorial, I left a TODO regarding whether to instruct the developer to designate that the module is to be activated _after_ the portal is initialized. Please let me know how or whether that is something we still want developers to do. This was mentioned in the original developer tutorial draft https://docs.google.com/document/d/1nXIm6dlxkvri143kYZLKf_PabxW8Y7Fo0odbFfIdXAE/edit#heading=h.ocyyrxgdh7lh.

Thanks!
Jim

### CODE REVIEW

**Comment by jhinkey** [comment]
FYI - I force pushed a correction to a paragraph referencing the schema versions used in the registration code.

**Comment by cgoncas** [comment]
Just started reviewing :) <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

File: develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-an-upgrade-process-for-your-application.markdown [hunk 1]
```diff
@@ -43,136 +154,180 @@ class modifies a column:
...
+registration  so that the upgrade framework records the module's `1.0.0` schema
+version in @product@'s `Release_` table. 
+
+**Important**: Modules that use Service Builder *should not* define a
+registration for their initial database schema version, as Service Builder
+already records their schema versions to @product@'s `Release_` table. Modules
+that don't use Service Builder, however, *should* define a registration for
+their initial schema. 
+
+The `DLWebUpgrade` registrator's second registration (from schema version
```
Comment (cgoncas) [inline develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-an-upgrade-process-for-your-application.markdown]: I don't know if this is the best example. The 0.0.1 step is something that only liferay provide. I think that it is better if we don't use a example with this step.

File: develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-an-upgrade-process-for-your-application.markdown [hunk 2]
```diff
@@ -43,136 +154,180 @@ class modifies a column:
...
+One important thing is that upgrades need to wait for portal initialization to perform their tasks.
 
-If your database schema is a different version than your module version, you can 
-specify it in the BND file. Add the `Require-SchemaVersion` property to your 
-`bnd.bnd` file to describe that the schema version of the module uses a 
-different version than the bundle version:
+You can achieve it by adding this setter method, with an OSGi reference to the portal life cycle.
 
-    Require-SchemaVersion: 1.0.2
+@Reference(target = ModuleServiceLifecycle.PORTAL_INITIALIZED, unbind = "-")
```
Comment (cgoncas) [inline develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-an-upgrade-process-for-your-application.markdown]: This should be not included, is an old reference.

File: develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-an-upgrade-process-for-your-application.markdown [hunk 3]
```diff
@@ -43,136 +154,180 @@ class modifies a column:
...
     }
 
-In this example the `com.liferay.calendar.service` module is upgraded from 
-version 0.0.1 to 1.0.0 in one step. Next it is upgraded incrementally from 
-version 1.0.1 to 1.0.4, using only one `UpgradeStep` for each schema version. 
-Finally, it is upgraded from version 1.0.4 to version 1.0.5, using a list of 
-`UpgradeSteps` with three steps in it: `UpgradeCalendarResource`, 
-`UpgradeCompanyId` and `UpgradeLastPublishDate`.
+The registrator declares itself to be an OSGi Component of service type
+`UpgradeStepRegistrator.class`. The `@Component` annotation registers the the
```
Comment (cgoncas) [inline develop/tutorials/articles/340-data-upgrades-and-verifiers/creating-an-upgrade-process-for-your-application.markdown]: There are a repetead 'the' here.

**Comment by cgoncas** [comment]
Hi @jhinkey 

I have some doubts:

**Comment by jhinkey** [comment]
cc\ @mwilliams2014

**Comment by jhinkey** [comment]
Hey Cristina! I force pushed changes to address your review comments. Thanks.

**Comment by cgoncas** [comment]
Just started reviewing :) <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Comment by cgoncas** [comment]
Pull request submitted to https://github.com/jhinkey/liferay-docs/pull/321. <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Pull request closed without merging**